isort/output.py

```diff
@@ -172,6 +166,9 @@ def sorted_imports(
         else:
             pending_lines_before = pending_lines_before or not no_lines_before
 
+    if config.ensure_newline_before_comments:
```

Comment:
Wasn't sure whether we want to insert an empty line before on a comment above the first import as well.
It looks a bit strange to me to have it there if there are not other imports before so I opted for not having an empty line in this special case.

In case you want it just move these 2 lines below the stripping of empty lines that follows next. I tested that it works there as well.